### ISSUE TITLE
Support Non Session Manager (NSM)

### ISSUE THREAD

**Issue by dbrgn** [calf-studio-gear/calf#32]
Would it be possible to add support for Non Session Manager to calfjackhost?

http://non.tuxfamily.org/wiki/Non%20Session%20Manager

The tool allows you to save and restore modular sessions in a very simple and nice way. A lot of applications have already implemented support for NSM: http://non.tuxfamily.org/wiki/ApplicationsSupportingNsm

**Comment by jrigg** [comment]
As a Non DAW user I'd say NSM support would be a very useful addition to calfjackhost.

**Comment by Houston4444** [comment]
+1

**Comment by trebmuh** [comment]
bumping and adding a +1 here

**Comment by riccardocagnasso** [comment]
+1

**Comment by erwan35** [comment]
+1

**Comment by falkTX** [comment]
+1

a C-header simple implementation is available at https://github.com/original-male/non/blob/master/nonlib/nsm.h that can be repurposed for calf

**Comment by diegodorado** [comment]
+1

**Comment by diegodorado** [comment]
Please,  this issue is open for 4 years. I am really happy with calf sidechain compressor, but not being able to save the session with NSM is making it useless to me... want can Ido to help with this?

**Comment by riccardocagnasso** [comment]
tbh calf works quite fine with nsm-proxy

**Comment by diegodorado** [comment]
It does work, but it doesnt saves the session...

**Comment by Houston4444** [comment]
@diegodorado  Try with [RaySession](https://github.com/Houston4444/RaySession) instead of Non Session Manager. However,save should works  with nsm-proxy with SIGUSR1 signal.

**Comment by riccardocagnasso** [comment]
> It does work, but it doesnt saves the session...

Yes it does. You have to pass a calf save file as command line parameter and use SIGUSR1 to signal to save. Fiddle a bit with it, you will get it to work.

**Comment by diegodorado** [comment]
Ohhh!! beautiful!! @riccardocagnasso @Houston4444 , you made my day!!
Works perfectly now, many thanks!
